src/node/desktop/src/main/preferences/file-preferences.ts

```diff
@@ -1,20 +1,45 @@
-import { UserDefaultTypes } from 'electron';
+/**
+ *
+ * file-preferences.ts
+ *
+ * Copyright (C) 2022 by RStudio, PBC
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+import { homedir } from 'os';
 import PropertiesReader from 'properties-reader';
-import { PlatformPreferences } from './preferences';
+import DesktopOptions from './desktop-options';
+import { preferenceKeys } from './preferences';
+
+const INI_FILE = 'desktop.ini';
 
-class FilePreferences implements PlatformPreferences {
+class FilePreferences extends DesktopOptions {
+  private iniFile: string; // existing location for some preferences
   private properties: PropertiesReader.Reader;
 
-  constructor(file: string) {
-    this.properties = PropertiesReader(file);
-  }
+  constructor() {
+    super();
+    if (process.platform === 'win32') {
```

Comment:
We should use the Xdg methods for determining these paths. These live in https://github.com/rstudio/rstudio/blob/main/src/node/desktop/src/core/xdg.ts, e.g. see `Xdg.userConfigDir()`, `Xdg.userDataDir()`.

Reply:
Good catch. I've changed it to use `userConfigDir()`. I've also added a check in case the file does not exist. It will fallback to the default font if no font selection is found.